appholder/src/main/java/com/android/identity/wallet/support/SecureAreaSupport.kt

```diff
@@ -0,0 +1,121 @@
+     */
+    @Composable
+    fun SecureAreaAuthUi(onUiStateUpdated: (newState: SecureAreaSupportState) -> Unit)
+
+    /**
+     * This function should create [SecureArea.KeyUnlockData] based on the incoming [Credential].
+     * It's implementation should decide on the mechanism that will do the unlocking (i.e. present
+     * a biometric prompts or other sort of UI), and the way the [SecureArea.KeyUnlockData] is created.
+     *
+     * The function is an extension on a [Fragment] due to the nature of Android and the navigation,
+     * so in case of rendering a UI specific for unlocking (like a Biometric Prompt, or a Dialog),
+     * there is a provided way to navigate using the [findNavController] function.
+     */
+    fun Fragment.unlockKey(
+        credential: Credential,
+        onKeyUnlocked: (unlockData: SecureArea.KeyUnlockData?) -> Unit,
+        onUnlockFailure: (wasCancelled: Boolean) -> Unit
+    )
+
+    /**
+     * Should return the current [SecureAreaSupportState] which is used by the composition
+     * when rendering the composable UI.
+     */
+    fun getSecureAreaSupportState(): SecureAreaSupportState
+
+    companion object {
+        fun getInstance(
+            context: Context,
+            secureAreaImplementationState: SecureAreaImplementationState
+        ): SecureAreaSupport {
+            return when (secureAreaImplementationState) {
+                is SecureAreaImplementationState.Android -> {
+                    val capabilities = AndroidKeystoreSecureArea.Capabilities(context)
+                    AndroidKeystoreSecureAreaSupport(capabilities)
+                }
+
+                is SecureAreaImplementationState.Software -> SoftwareKeystoreSecureAreaSupport()
+            }
+        }
+
+        fun getInstance(
+            context: Context,
+            credential: Credential
+        ): SecureAreaSupport {
+            return when (credential.credentialSecureArea) {
+                is AndroidKeystoreSecureArea -> {
+                    val capabilities = AndroidKeystoreSecureArea.Capabilities(context)
+                    AndroidKeystoreSecureAreaSupport(capabilities)
+                }
+
+                is SoftwareSecureArea -> SoftwareKeystoreSecureAreaSupport()
+
+                else -> SecureAreaSupportNull()
+            }
+        }
+
+        fun createCredentialStore(
+            context: Context,
+            keystoreEngineRepository: SecureAreaRepository
+        ): CredentialStore {
+            val storageDir = PreferencesHelper.getKeystoreBackedStorageLocation(context)
+            val storageEngine = AndroidStorageEngine.Builder(context, storageDir).build()
+
+            val androidKeystoreSecureArea = AndroidKeystoreSecureArea(context, storageEngine)
+            val softwareSecureArea = SoftwareSecureArea(storageEngine)
+
+            keystoreEngineRepository.addImplementation(androidKeystoreSecureArea)
+            keystoreEngineRepository.addImplementation(softwareSecureArea)
```

Comment:
I don't think this is the right place to populate the `SecureArea` implementations the application wishes to use. I think that belongs in `HolderApp.onCreate()`.